src/pybind/decoder/decoder_wrappers_pybind.cc

```diff
@@ -0,0 +1,72 @@
+// pybind/decoder/decoder_wrappers_pybind.cc
+
+// Copyright 2020   Mobvoi AI Lab, Beijing, China
+//                  (author: Fangjun Kuang, Yaguang Hu, Jian Wang)
+
+// See ../../../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABLITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#include "decoder/decoder_wrappers_pybind.h"
+
+#include "decoder/decoder-wrappers.h"
+
+using namespace kaldi;
+
+namespace {
+
+template <typename FST>
+void pybind_decode_utterance_lattice_faster_impl(
+    py::module& m, const std::string& func_name,
+    const std::string& func_help_doc = "") {
```

Comment:
Can you explain to me why this func_help_doc string is passed in and not just hardcoded?

Reply:
I find that the current approach is more flexible.

If it is hardcoded, then `fst::StdFst` and `fst::GrammarFst`
will share the same help documentation.

If later we want to add a new type fst, it will use the same
help message and we are not able to provide a customized help message.